bson/unmarshal_test.go

```diff
@@ -166,3 +167,114 @@ func TestCachingDecodersNotSharedAcrossRegistries(t *testing.T) {
 		assert.Equal(t, int32(-1), *second.X, "expected X value to be -1, got %v", *second.X)
 	})
 }
+
+type expectedResponse struct {
+	DefinedField string
+}
+
+func unmarshalExtJSONHelper(extJSON string) (*expectedResponse, error) {
```

Comment:
nit: I understand the purpose of this, but functions with generic names like this can cause issues for future authors. Someone trying to define a similar function for a different type in a test would run into trouble because the name is already taken here, so we'd end up with more convoluted names.

I'd recommend defining both the `expectedResponse` type and this function within the test function so their scope is limited. I have a small preference for this approach because it makes it very obvious that this type is tied to a specific test:

```
func TestUnmarshalExtJSONWithUndefinedField(t *testing.T) {
    type expectedResponse struct {
        DefinedField string
    }

    unmarshalExpectedResponse := func(t *testing.T, extJSON string) {
        t.Helper()
        err := UnmarshalExtJSON([]byte(extJSON), false, &responseDoc)
        assert.Nil(t, err, "UnmarshalExtJSON error: %v", err)
    }
}
```
Note that I changed the function slightly to give it a different name and have it directly accept a `testing.T` instance so it can fail the test if unmarshalling errors. The `t.Helper()` call at the beginning makes it so stack traces for a failed test will leave this function out (see https://gist.github.com/divjotarora/7ea6f953fd3cd7d21f0bd170e319768b for an example).

Reply:
That sounds good! Moved them within `TestUnmarshalExtJSONWithUndefinedField()`, added the references to `testing.T` and changed the name.